bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,723 @@
+        }
+
+        boolean success = true;
+        synchronized (sequenceLock) {
+            if (!connector.isConnected()) {
+                logger.debug("Command {} from channel {} is ignored: connection not established", command, channel);
+                return;
+            }
+
+            Stream<String> zoneStream = MonopriceAudioZone.VALID_ZONES.stream();
+            try {
+                switch (channelType) {
+                    case CHANNEL_TYPE_POWER:
+                        if (command instanceof OnOffType && command == OnOffType.ON) {
+                            connector.sendCommand(zone, MonopriceAudioCommand.POWER_ON);
+                            ZONE_DATA_MAP.get(zone.getZoneId()).setPower(ON);
+                        } else if (command instanceof OnOffType && command == OnOffType.OFF) {
+                            connector.sendCommand(zone, MonopriceAudioCommand.POWER_OFF);
+                            ZONE_DATA_MAP.get(zone.getZoneId()).setPower(OFF);
+                        }
+                        break;
+                    case CHANNEL_TYPE_SOURCE:
+                        if (command instanceof DecimalType) {
+                            int value = ((DecimalType) command).intValue();
+                            if (value >= ONE && value <= MAX_SRC) {
+                                logger.debug("Got source command {} zone {}", value, zone);
+                                connector.sendCommand(zone, MonopriceAudioCommand.SOURCE, value);
+                                ZONE_DATA_MAP.get(zone.getZoneId()).setSource(String.format("%02d", value));
+                            }
+                        }
+                        break;
+                    case CHANNEL_TYPE_VOLUME:
+                        if (command instanceof PercentType) {
+                            int value = (int) Math
+                                    .round(((PercentType) command).doubleValue() / 100.0 * (MAX_VOLUME - MIN_VOLUME))
+                                    + MIN_VOLUME;
+                            logger.debug("Got volume command {} zone {}", value, zone);
+                            connector.sendCommand(zone, MonopriceAudioCommand.VOLUME, value);
+                            ZONE_DATA_MAP.get(zone.getZoneId()).setVolume(String.format("%02d", value));
+                        }
+                        break;
+                    case CHANNEL_TYPE_MUTE:
+                        if (command instanceof OnOffType && command == OnOffType.ON) {
+                            connector.sendCommand(zone, MonopriceAudioCommand.MUTE_ON);
+                            ZONE_DATA_MAP.get(zone.getZoneId()).setMute(ON);
+                        } else if (command instanceof OnOffType && command == OnOffType.OFF) {
+                            connector.sendCommand(zone, MonopriceAudioCommand.MUTE_OFF);
+                            ZONE_DATA_MAP.get(zone.getZoneId()).setMute(OFF);
+                        }
```

Comment:
I feel like all these on/off command handlers could be simplified to a single function.
```
private void handleOnOff(Command command, MonopriceAudioZone zone, MonopriceAudioCommand onCommand, MonopriceAudioCommand offCommand, BiConsumer<MonopriceAudioZoneDTO, String> adjuster){
    if( command == OnOffType.ON){
        connector.sendCommand(zone, onCommand);
        adjuster.accept(ZONE_DATA_MAP.get(zone.getZoneId()), ON);
    } else if (command == OnOffType.OFF){
        connector.sendCommand(zone, offCommand);
        adjuster.accept(ZONE_DATA_MAP.get(zone.getZoneId()), OFF);
    }
}
```
```suggestion
                        handleOnOff(command, zone, MonopriceAudioCommand.MUTE_ON, MonopriceAudioCommand.MUTE_OFF, MonopriceAudioZone::setMute);
```

Reply:
I condensed them all with a ternary.